### DISCUSSION TITLE
container-diff integration

### DISCUSSION THREAD

**Discussion by BigMichi1** [containrrr/watchtower#1242] [Q&A] [answered]
Hello,

I'm currently trying to integrate https://github.com/GoogleContainerTools/container-diff as a `post-update` hook but I'm not sure how to get the 2 digests of the images (the one from before the update and the one after the update).

the goal is to send an email containing the difference in the images (in my case files that are different) to get a little bit more information on what might have changed when a container was updated.

**Comment by piksel** [comment, marked as answer]
If you pass the `--notification-report` argument to watchtower, it will send a notification including the old and the new image hashes. You can customize the template as well to suite your needs, the default format for the "report" notifications is:
https://github.com/containrrr/watchtower/blob/f79e4b5435baf8cdf6f69281ae11a7215e17f320/pkg/notifications/shoutrrr.go#L21-L42

You would still need to pipe that notification through container-diff, but that is the closest thing I can think of right now. There is no way to get the image hashes in the post-update afaik.
